ElrondNetwork/elrond-go
# Prevent parallel snapshots

## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- If a snapshot takes more than one epoch, another snapshot will start in parallel. In extreme cases, this will lead to an IO bottleneck

  
## Proposed Changes
- Add `isSnapshotInProgress` flag in order to prevent this kind of behaviour, and skip a snapshot if another one is running.
- Remove duplicated code from AccountsDB and PeerAccountsDB


## Testing procedure
- Create another branch, and add a time.Sleep(1h) to simulate a snapshot that takes 1h. No other snapshot should start during this time.
- Another normal testing procedure to check that nothing has changed in normal cases.
